ocs_ci/utility/kms.py

```diff
@@ -1797,7 +1802,284 @@ def cleanup(self):
 
 
-kms_map = {"vault": Vault, "hpcs": HPCS, "kmip": KMIP}
+class AzureKV(KMS):
+    """
+    Represents an Azure Key Vault implementation of KMS.
+    """
+
+    def __init__(self, namespace=config.ENV_DATA["cluster_namespace"]):
+        super().__init__(constants.AZURE_KV_PROVIDER_NAME)
+        self.namespace = namespace
+        self.kms_provider = constants.AZURE_KV_PROVIDER_NAME
+        self.azure_kms_connection_name = (
+            f"azure-kv-conn-{gen_alphanumeric(length=5).lower()}"
+        )
+        azure_auth = config.AUTH.get("azure_auth")
+        self.azure_kv_name = azure_auth.get("AZURE_KV_NAME")
+        self.azure_kv_certificate = azure_auth.get("AZURE_CERTIFICATE")
+        self.vault_url = azure_auth.get("AZURE_KV_URL")
+        self.vault_client_id = azure_auth.get("AZURE_KV_CLIENT_ID")
+        self.vault_tenant_id = azure_auth.get("AZURE_KV_TENANT_ID")
+        self.vault_cert_path = self._vault_cert_path()
+
+        self.conn_data = {
+            "KMS_PROVIDER": self.kms_provider,
+            "KMS_SERVICE_NAME": self.azure_kms_connection_name,
+            "AZURE_CLIENT_ID": self.vault_client_id,
+            "AZURE_VAULT_URL": self.vault_url,
+            "AZURE_TENANT_ID": self.vault_tenant_id,
+        }
+
+    def deploy(self):
+        """
+        This Function will create the Azure KV connection details in the ConfigMap.
+        """
+        if not config.ENV_DATA.get("platform") == "azure":
+            raise VaultDeploymentError(
+                "Azure_KV deployment only supports on Azure platform."
+            )
+
+        self.create_azure_kv_csi_kms_connection_details()
+        if config.ENV_DATA.get("encryption_at_rest"):
+            self.create_azure_kv_ocs_csi_kms_connection_details()
+
+    def post_deploy_verification(self):
+        """
+        Post Deploy Verification For Azure Key Vault.
+        """
+        if config.ENV_DATA.get("encryption_at_rest"):
+            if not self.verify_osd_keys_present_on_azure_kv():
+                raise ValueError("OSD keys Not present on Azure Key Vault.")
+            logger.info("OSD Keys Are present on Azure Key Vault.")
+
+    def is_azure_kv_connection_exists(self):
+        """
+        Checks if the Azure KV connection exists in the ConfigMap
+        """
+
+        csi_kms_configmap = ocp.OCP(
+            kind="ConfigMap",
+            resource_name=constants.VAULT_KMS_CSI_CONNECTION_DETAILS,
+            namespace=self.namespace,
+        )
+
+        if not csi_kms_configmap.is_exist():
+            raise ValueError(
+                f"ConfigMap {csi_kms_configmap.resource_name} Not found in the namespace {self.namespace}"
+            )
+
+        if self.azure_kms_connection_name not in csi_kms_configmap.data["data"]:
+            raise ValueError(
+                f"Azure Key vault connection {self.azure_kms_connection_name} not exists."
+            )
+
+        return True
```

Comment:
the function would either raise an exception or return True. It can never return False.
In order to keep the consistency of returning a bool value, suggesting to catch the exception and return False in case Azure KV connection is not found. And throw an exception from outside the function if needed.

Reply:
Done